Remove Authentication from names where possible

HaoK opened the pull request:
Fixes https://github.com/aspnet/Security/issues/428

HaoK commented:
cc @Tratcher

Tratcher commented on samples/CookieSample/Startup.cs:
```diff
@@ -28,8 +28,8 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
-                    var user = new ClaimsPrincipal(new ClaimsIdentity(new[] { new Claim(ClaimTypes.Name, "bob") }, CookieAuthenticationDefaults.AuthenticationScheme));
-                    await context.Authentication.SignInAsync(CookieAuthenticationDefaults.AuthenticationScheme, user);
+                    var user = new ClaimsPrincipal(new ClaimsIdentity(new[] { new Claim(ClaimTypes.Name, "bob") }, CookieDefaults.AuthenticationScheme));
+                    await context.Authentication.SignInAsync(CookieDefaults.AuthenticationScheme, user);
```
CookieMiddlewareDefaults?

HaoK replied:
We should keep it Authentication if its unclear, so I can add that back for Cookies

Tratcher commented on src/Microsoft.AspNet.Authentication.OpenIdConnect/Events/IOpenIdConnectAuthenticationEvents.cs:
```diff
@@ -6,7 +6,7 @@
-    /// Specifies events which the <see cref="OpenIdConnectAuthenticationMiddleware" />invokes to enable developer control over the authentication process.
+    /// Specifies events which the <see cref="OpenIdConnectMiddleware" />invokes to enable developer control over the authentication process.
     /// </summary>
     public interface IOpenIdConnectAuthenticationEvents
```
IOpenIdConnectEvents?

Tratcher commented on src/Microsoft.AspNet.Authentication.OpenIdConnect/Events/OpenIdConnectAuthenticationEvents.cs:
```diff
@@ -7,7 +7,7 @@
-    /// Specifies events which the <see cref="OpenIdConnectAuthenticationMiddleware" />invokes to enable developer control over the authentication process.
+    /// Specifies events which the <see cref="OpenIdConnectMiddleware" />invokes to enable developer control over the authentication process.
     /// </summary>
     public class OpenIdConnectAuthenticationEvents : IOpenIdConnectAuthenticationEvents
```
OpenIdConnectEvents?

HaoK replied:
Good catch yeah I missed events in OAuth and OIDC, nested folders that were collapsed

Tratcher commented on src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectMethod.cs:
```diff
@@ -4,7 +4,7 @@
     /// redirect the user agent to the identity provider.
     /// </summary>
-    public enum OpenIdConnectAuthenticationMethod
+    public enum OpenIdConnectMethod
```
@PinpointTownes What do you think? This one sounds a little off.

HaoK replied:
OpenIdConnectBehavior?

HaoK replied:
OpenIdConnectRedirectBehavior?

PinpointTownes replied:
OpenIdConnectRedirectBehavior seems fine. 

That said, removing Authentication from all the middleware except cookies is really weird. For consistency and to avoid collisions (GoogleOptions or FacebookOptions are far too generic and may result in duplicate names if you use a FB/Google client in Startup.cs), I'd keep Authentication everywhere.

Tratcher replied:
:watch:

Tratcher commented on test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectAuthenticationHandlerForTestingAuthenticate.cs:
```diff
@@ -15,7 +15,7 @@ namespace Microsoft.AspNet.Authentication.Tests.OpenIdConnect
     ///  Allows for custom processing of ApplyResponseChallenge, ApplyResponseGrant and AuthenticateCore
     /// </summary>
-    public class OpenIdConnectAuthenticationHandlerForTestingAuthenticate : OpenIdConnectAuthenticationHandler
+    public class OpenIdConnectAuthenticationHandlerForTestingAuthenticate : OpenIdConnectHandler
```
Missed a few here.

HaoK replied:
Ugh, on a side note, at some point are you going to nuke these classes?  They really don't seem necessary, the middleware exists to replace the logger (instead of just putting the right logger in the container)

Tratcher replied:
Eventually, but that's unrelated to this pass.

Tratcher commented:
Most of them seem fine but this is strange to apply to Cookies. It's ambiguous between HTTP cookies and cookie authentication. I think you should revert most/all of the CookieAuth renames.

HaoK commented:
That's reasonable, so we'll keep CookieAuthentication as there's Http.Cookie already.  I'll revert on merge/squash as its easier than trying to undo them manually

HaoK commented:
Updated PR with cookie names reverted except for the internal Handler class

PinpointTownes commented:
Curious: what's the benefit of removing `Authentication` from all these classes?
IMHO, the new names look so artificial :scream:

HaoK commented:
Authentication is already in the namespace. So it's just extra length that adds little meaning.

Tratcher commented on samples/SocialSample/Startup.cs:
```diff
@@ -45,6 +45,8 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
                 options.LoginPath = new PathString("/login");
             });
 
+            //app.UseFacebookAuthentication(app.GetOptions<FacebookOptions>()));
```
?

HaoK replied:
Oops yeah I'll remove

On Sep 11, 2015, at 10:23 AM, Chris R <notifications@github.com<mailto:notifications@github.com>> wrote:


In samples/SocialSample/Startup.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2faspnet%2fSecurity%2fpull%2f441%23discussion_r39294307&data=01%7c01%7chaok%40microsoft.com%7c620df78cee514b960a8708d2bacdbcf3%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=oTqI7xNpED2vKfnhlBqVh7SD%2bgIPSGbyXBLmBGMb4cQ%3d>:

> @@ -45,6 +45,8 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
>                  options.LoginPath = new PathString("/login");
>              });
>
> +            //app.UseFacebookAuthentication(app.GetOptions<FacebookOptions>()));


?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2faspnet%2fSecurity%2fpull%2f441%2ffiles%23r39294307&data=01%7c01%7chaok%40microsoft.com%7c620df78cee514b960a8708d2bacdbcf3%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=WepOsGfG1RAOONJhRGt2briCeeZz5YgKaowTzK2cP7g%3d>.

Tratcher commented on src/Microsoft.AspNet.Authentication.Cookies/CookieHandler.cs:
```diff
@@ -16,7 +16,7 @@
 namespace Microsoft.AspNet.Authentication.Cookies
 {
-    internal class CookieAuthenticationHandler : AuthenticationHandler<CookieAuthenticationOptions>
+    internal class CookieHandler : AuthenticationHandler<CookieAuthenticationOptions>
```
No, revert this one too.

Tratcher commented on src/Microsoft.AspNet.Authentication.Cookies/Events/CookieAuthenticationEvents.cs:
```diff
@@ -54,30 +58,30 @@ public class CookieAuthenticationEvents : ICookieAuthenticationEvents
         /// </summary>
         /// <param name="context"></param>
-        public virtual void ResponseSignIn(CookieResponseSignInContext context) => OnResponseSignIn(context);
+        public virtual Task ResponseSignIn(CookieResponseSignInContext context) => OnResponseSignIn(context);
```
You changed the return value here, but you didn't add any await's at the call sites in the Handler.

HaoK replied:
I actually didn't mean for any async changes to bleed into this pr yet. I will revert those. And do a new pr for those stand alone

Tratcher commented:
:watch:

Tratcher commented:
@Eilon for style.

HaoK commented:
Updated, renamed to OpenIdConnectRedirectBehavior, added awaits to Cookie, I'm going to do the async changes in a subbranch for a cleaner PR, but I'm going to merge all of the breaking changes to dev together when they are all ready

Eilon commented:
@Tratcher what style question did you have? I don't immediately see anything...

Tratcher commented:
@Eilon Just the overall naming esthetic.

Eilon commented:
Ah, yeah, looks good to me. I looked at all the filenames only - I'm assuming the type names match the filenames.

@HaoK please open an Announcements issue specifying at least the key changes.

Tratcher commented:
:shipit:

HaoK commented:
e8090a3176de5e6fc84be829be3a98f1e5ee8a5d

Eilon commented:
@HaoK please remember to open an issue here: https://github.com/aspnet/Announcements/issues/
